docs: update docs for guest wishlist (#354)
vuestorefront/bigcommerce: docs/composables/use-guest-wishlist.md

## Features

The `useGuestWishlist` composable is used to:

- create guest wishlist and store it in the `localStorage`,
- load guest wishlist from the `localStorage`,
- add products to the guest wishlist,
- remove products from the guest wishlist.

## API

- `wishlist: Wishlist` - a main data object.

> - `load` - function for getting wishlist from the `localStorage` or creating a new one if it doesn't exist. This method populates the `wishlist` object with the result.

Reviewer: ```suggestion
- `load` - function for getting wishlist from the `localStorage` or creating a new one if it doesn't already exist. This method populates the `wishlist` object with the result.
```

> - `addItem` - function for adding products to the guest wishlist. This method populates the `wishlist` object with the updated result. This method accepts a single object with the following properites:

Reviewer: ```suggestion
- `addItem` - function for adding products to the guest wishlist. This method populates the `wishlist` object with the updated result and accepts a single object parameter with the following properties:
```

>   - `product: Product` - See [Product](../reference/api/bigcommerce-api.product.md) for more information.

Reviewer: ```suggestion
  - `product: Product` - See the [Product](../reference/api/bigcommerce-api.product.md) interface for more information.
```

> - `removeItem` - function for removing a item that currently is in the guest wishlist. This method populates the `wishlist` object with the updated result. This method accepts a single object with following properites:

Reviewer: ```suggestion
- `removeItem` - function for removing a single item from the guest wishlist. This method populates the `wishlist` object with the updated result and accepts a single object parameter with the following properties:
```

>   - `product: WishlistItem` - See [WishlistItem](../reference/api/bigcommerce.wishlistitem.md) for more information.

Reviewer: ```suggestion
  - `product: WishlistItem` - See tthe [WishlistItem](../reference/api/bigcommerce.wishlistitem.md) interface for more information.
```

- `clear` - a function that removes all products from the wishlist. This method populates the `wishlist` object with the updated result.

- `loading: boolean` - a reactive object containing information about loading state of the wishlist.

> - `error: UseWishlistErrors` - reactive object containing the error message, if some methods failed for any reason.

Reviewer: ```suggestion
- `error: UseWishlistErrors` - reactive object containing the error message if any of the methods above failed for any reason. See [UseWishlistErrors](ADD PATH) for more information.
```

Change `ADD PATH` for actual link 😊
